### ISSUE TITLE
Importance weights.

### ISSUE THREAD

**Issue by jake-foxy** [GarrettJenkinson/condor_tensorflow#4]
I had a question about the importance weights code below that was in one of the tutorial docs.

```
Importance weights customization
A quick example to show how the importance weights can be customized.
model = create_model(num_classes = NUM_CLASSES)
model.summary()
# We have num_classes - 1 outputs (cumulative logits), so there are 9 elements
# in the importance vector to customize.
importance_weights = [1., 1., 0.5, 0.5, 0.5, 1., 1., 0.1, 0.1]
loss_fn = condor.SparseCondorOrdinalCrossEntropy(importance_weights = importance_weights)
model.compile(tf.keras.optimizers.Adam(lr = learning_rate), loss = loss_fn)
history = model.fit(dataset, epochs = num_epochs)
```
My problem:

I have 5 classes, with underrepresentation of say the first and lass class. I want to use weights to assign higher importance to the underrepresented classes. In a dense layer with n(classes) == n(output_layers), the vector would look like.

[1,0.5,0.5,0.5,1] 

With the CONDOR, using  num_classes - 1 output layers, is it still possible to assign higher weights to underrepresented classes? 

I don't understand how to relate the N-1 output layers weights to the original weights where n(classes) == n(output_layers).

Any feedback is appreciated.

**Comment by jake-foxy** [comment]
As a follow up, I have also tried :

```
`history = new_model.fit(
    train_generator,
    validation_data=val_generator,
    validation_steps=STEP_SIZE_VALID,
    steps_per_epoch=nbatches_train,
    epochs=int(MODEL_PARAMETERS["num_epochs"]),
    workers=8,
    shuffle=True,
    verbose=1,
    callbacks=[mc],
    class_weight = class_weights`
```

Where class_weight is a dictionary of weights equal to the number of classes. Is this appropriate? It runs successfully and has an effect on the model output. I am unsure why the tutorial docs suggest the above weights (N-1) within the loss, instead of providing  class weights in the fit call.

**Comment by GarrettJenkinson** [comment]
Your use of `class_weight` is appropriate for your situation. This approach assigns to each training instance a weight based on the class, thereby overweighting the minority class.

The `importance_weights` route refers to the weights assigned to each binary subtask. This impacts each and every training sample equally. It is most appropriate when your problem has variable importance between the classes themselves. So as an example, if you had four classes going "benign", "likely benign", "likely pathogenic", "pathogenic", you might say that it VERY important that a benign is not misclassified as pathogenic (or vice versa), but less important to distinguish "likely benign" from "benign". In this case, you would want your weights to look something like [0.5,1.0,0.5] to place twice the weight on the subtask "greater than likely benign". 

Hope this helps, let me know if you have any other questions on usage here.
